packages/aws-rfdk/docs/upgrade/upgrading-0.37.md

```diff
@@ -0,0 +1,41 @@
+# Upgrading to RFDK v0.37.x or Newer
+
+Starting in RFDK v0.37.0, the default for TLS between the render queue and its clients, which is configured using the `RenderQueueExternalTLSProps` interface that the `RenderQueue` construct takes as a part of its constructor props, is now set to be enabled.
+
+## Upgrading Farms Already Using TLS
+
+If you are already setting fields on the `RenderQueueExternalTLSProps` for the Render Queue, no action is required. Redeploying your render farm after upgrading your version of RFDK should have no effect.
+
+## Upgrading Farms Not Using TLS
+
+### RenderQueue Changes
+
+Versions of RFDK prior to 0.37.0 had internal TLS between the load balancer and its backing services on by default. This is configurable with the `internalProtocol` field on the `RenderQueueTrafficEncryptionProps` interface. This default was left as-is, so upgrading RFDK will have no effect on the protocol those backing services were already using and they will not need to be replaced. The TLS being enabled by default is between the listener on the load balancer and any Deadline clients that are connecting to it, which is configurable with the `externalProtocol` property on the `RenderQueueTrafficEncryptionProps` interface.
+
+There will be a few new constructs deployed to your farm:
+1. A `PrivateHostedZone` will be created if you do not supply your own. We set the default domain to `aws-rfdk.com`, which we have registered and suggest that you use if you do not have your own registered domain. [RFC 6762](https://datatracker.ietf.org/doc/html/rfc6762#appendix-G) recommends against using any unregistered top-level domains.
+1. A self-signed X509 certificate will be generated using OpenSSL and that will then be used to sign a certificate that the Render Queue will use for TLS. Specifically, the certificate will be passed to the Application Listener for the Application Load Balancer that the Render Queue creates. Additional details about how RFDK uses TLS can the built-in certificate management can be found in the developer guide for [Encryption in transit](https://docs.aws.amazon.com/rfdk/latest/guide/security-encrypt-in-transit.html).
+
+These new constructs will require the Render Queue load balancer's listener to need replacing, but the load balancer itself and the backing services it redirects traffic to will not need to be changed.
+
+### WorkerInstanceFleet Changes
```

Comment:
Good catch to think about this and the CDK ["deadly embrace" issue](https://github.com/aws/aws-cdk/issues/7602) that would arise.

I'm curious what happens with the [`ConfigureSpotEventPlugin` construct](https://docs.aws.amazon.com/rfdk/api/latest/docs/aws-rfdk.deadline.ConfigureSpotEventPlugin.html) in this upgrade scenario. Has this been tested? My theory is that it would unfold as follows:

1. The deployment would replace the RQ ALB listener
2. The deployment would proceed to deploy the update to the `Custom::RFDK_ConfigureSpotEventPlugin` which would reconfigure the Deadline Spot Event Plugin with the new `RenderQueue` configuration. Changing the Spot Event Plugin configuration does not immediately apply changes to the fleets that it manages. The Spot Event Plugin evaluates its configuration and applies spot fleet request changes [at the next Deadline house cleaning event](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/event-spot.html#lifecycle).
3. Fork depending on if [the ResourceTracker is enabled](https://docs.aws.amazon.com/rfdk/api/latest/docs/aws-rfdk.deadline.SpotEventPluginSettings.html#enableresourcetracker):
    1.  **If enabled**, the existing `SpotEventPluginFleet`s would lose their connection to the RQ. There would be a race condition that depends on the Deadline Resource Tracker's configured [fleet fault tolerance and thresholds](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/resource-tracker-overview.html#fleet-fault-tolerance-and-termination-threshold). The Resource Tracker would start by rebooting unhealthy instances and eventually mark the fleet as unhealthy and would [need to be reset](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/resource-tracker-troubleshooting.html#unable-to-start-new-instances).
    2.  **If disabled**, the new Deadline Spot Event Plugin configuration would be applied to the existing spot fleet request at the next Deadline house cleaning event and Spot Workers would resume operation.

Reply:
Thanks for looking into this. Looks like it wasn't very straight-forward of an upgrade process, so it's helpful that you've tested and have a documented process for doing this cleanly. 👏